src/test/java/core/basesyntax/service/ParserServiceTest.java

```diff
@@ -0,0 +1,102 @@
+package core.basesyntax.service;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.model.FruitModel;
+import core.basesyntax.service.impl.ParserServiceImpl;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ParserServiceTest {
+    private static final FruitModel.Operation BALANCE = FruitModel.Operation.BALANCE;
+    private static final FruitModel.Operation PURCHASE = FruitModel.Operation.PURCHASE;
+    private static final FruitModel.Operation SUPPLY = FruitModel.Operation.SUPPLY;
+    private static final FruitModel.Operation RETURN = FruitModel.Operation.RETURN;
+    private static final String BANANA = "banana";
+    private static final String APPLE = "apple";
+    private static List<String> list;
+    private static List<FruitModel> fruitModels;
+    private static ParserService parserService;
+
+    @BeforeClass
+    public static void beforeAll() {
+        list = new ArrayList<>();
+        fruitModels = new ArrayList<>();
+        parserService = new ParserServiceImpl();
+    }
+
+    @Before
+    public void init() {
+        list = Stream.of("type,fruit,quantity", "b,banana,20", "b,apple,100", "s,banana,100",
+                        "p,banana,13", "r,apple,10", "p,apple,20", "p,banana,5", "s,banana,50")
+                .collect(Collectors.toList());
+        fruitModels = Stream.of(
+                        new FruitModel(BALANCE, BANANA, 20),
+                        new FruitModel(BALANCE, APPLE, 100),
+                        new FruitModel(SUPPLY, BANANA, 100),
+                        new FruitModel(PURCHASE, BANANA, 13),
+                        new FruitModel(RETURN, APPLE, 10),
+                        new FruitModel(PURCHASE, APPLE, 20),
+                        new FruitModel(PURCHASE, BANANA, 5),
+                        new FruitModel(SUPPLY, BANANA, 50))
+                .collect(Collectors.toList());
+    }
+
+    @Test
+    public void parseService_ok() {
+        List<FruitModel> actualParseService = parserService.parseData(list);
```

Comment:
```suggestion
        List<FruitModel> parseResult = parserService.parseData(list);
```
or something like this